src/esi/Esi.cc

```diff
@@ -1188,7 +1188,7 @@ ESIContext::addLiteral (const char *s, int len)
     assert (len);
     debugs(86, 5, "literal length is " << len);
     /* give a literal to the current element */
-    Must(parserState.stackdepth < 10);
+    Must2(parserState.stackdepth < 10, "Too many nested elements");
```

Comment:
Why check `parserState.stackdepth` that this method code does not use itself? Perhaps this check belongs to those addElement() implementations that actually increment `parserState.stackdepth` and/or use it as an array index?

Reply:
The exception string generated by these Must is displayed publicly. That means the caller / location of the Must itself matters for admin debugging how exactly the ESI response is broken.

Reply:
I agree that Must() location matters, for many reasons.

This particular Must() is wrong because it checks something that the surrounding code does not care about. If you are using the wrong location because it relays some useful information to the debugging admin, then, most likely, there is a better way to relay that information. In other words, I am not convinced that the best way to relay ESI response breakage details is to put a `Must(foo)` in a place where `foo` is unused.

Reply:
resolved with the the exception model